src/include/session.h

```diff
@@ -53,6 +53,12 @@ typedef TAILQ_HEAD(__wt_cursor_list, __wt_cursor) WT_CURSOR_LIST;
 /* Maximum number of buckets to visit during a regular cursor sweep. */
 #define WT_SESSION_CURSOR_SWEEP_MAX 64
 
+/* Invalid session ID. */
+#define WT_SESSION_ID_INVALID 0xffffffff
+
+/* A fake session ID for when we need to refer to a session that is actually NULL. */
+#define WT_SESSION_ID_NULL 0xfffffffe
```

Comment:
I was thinking the session is actually null in this case but apparently I was wrong. Who set the session id to this value and why we don't have this definition before the pr?

Reply:
I just added these two special session IDs that I ran into:

- given that the session ID is defined as an index into an array, 0 is thus considered a valid session ID
- we actually call lock/unlock in two places with session being NULL, for which I also needed to define an ID value for the purposes of the locks

I suppose we didn't need them before! :)

If you're curious, you can find the NULL sessions being used here:

- src/conn/conn_cache_pool.c:402:    `__wt_spin_lock(NULL, &cp->cache_pool_lock);`
- src/conn/conn_api.c:2717:    `__wt_spin_lock(NULL, &__wt_process.spinlock);`

Reply:
I get it now. Thanks.